### ISSUE TITLE
understanding the dataloader procedure 

### ISSUE THREAD

**Issue by seyeeet** [icoz69/DeepEMD#12]
Thanks so much for providing the code of your paper.
I am new to FSL and trying to understand the training and data loader process. I understand the concept of `support` and `query` as well as `shot` and `way`. But I'm still confused about how you make the data loader and appreciate if you can explain it. I also read the paper several times but it didn't work to understand these details. I really appreciate your help in advance.

1) In the case of  `1-shot 5-way`, what does `query=15` mean? how should we choose the number for query? does it need to be some specific number in agreement with previous works or it is arbitrary and kinda like hyperparameter?

in the `cub` dataset, we have `5892` images. and I see the data loader generate batch of `80x3x84x84` for input and the output is in forms of `80x640x5x5`: 

2) I don't understand where the `80` comes from at the first point, and then in the output where the `640` comes from?

3) later I see we have `data_shot` and `logits`, the size of `logit` is `75x5`, can you please explain what `75` represent? I understand that it is `15*5=75` but I could not understand what it represents.

Thanks in advance for your help.

**Comment by icoz69** [comment]
hello
1.query=15 means the number of test images each class in this mini-task (an episode). it can be changed. since the evaluation is by repeatedly sampling tasks, larger query number and sampling times can make the average result more stable and convincing. for example, if you have large gpu memory, you can set a larger query size for training and eval.
2. 80 is the number of query images plus the number of support images, i.e. 5 way * (1 support image + 15 query image). 640 is the dim of the feature embedding.
3.75 is the predicted scores of 75 query images. since each class has 15 query images, and you have 5 class, you have 75 testing images totally.
